packages/framer-motion/src/projection/node/create-projection-node.ts

```diff
@@ -1417,30 +1417,34 @@ export function createProjectionNode<I>({
             // If there's no detected rotation values, we can early return without a forced render.
             let hasRotate = false
 
-            // Keep a record of all the values we've reset
+            /**
+             * An unrolled check for rotation values. Most elements don't have any rotation and
+             * skipping the nested loop and new object creation is 50% faster.
+             */
+            const { latestValues } = visualElement
+            if (
+                latestValues.rotate ||
+                latestValues.rotateX ||
+                latestValues.rotateY ||
+                latestValues.rotateZ
+            ) {
+                hasRotate = true
+            }
+
+            // If there's no rotation values, we don't need to do any more.
+            if (!hasRotate) return
+
             const resetValues = {}
 
             // Check the rotate value of all axes and reset to 0
             for (let i = 0; i < transformAxes.length; i++) {
-                const axis = transformAxes[i]
-                const key = "rotate" + axis
-
-                // If this rotation doesn't exist as a motion value, then we don't
-                // need to reset it
-                if (!visualElement.getStaticValue(key)) {
-                    continue
-                }
-
-                hasRotate = true
+                const key = "rotate" + transformAxes[i]
 
                 // Record the rotation and then temporarily set it to 0
-                resetValues[key] = visualElement.getStaticValue(key)
+                resetValues[key] = latestValues[key]
                 visualElement.setStaticValue(key, 0)
```

Comment:
Should we check for `latestValues[key]` to be `0` here too?
Or wouldn't that lead to much preformance benifit?

Reply:
Ah yeah it's true we don't need to do anything for those values